### PR TITLE
Add WeChat Receiver type to AlertmanagerConfig CRD

### PR DESCRIPTION
**Pull request by zwpaper**
fix https://github.com/prometheus-operator/prometheus-operator/issues/3537

btw, I updated the user guide doc to use AlertmanagerConfig CRD

### CODE REVIEW

File: Documentation/user-guides/alerting.md
Comment (grdryn) [inline Documentation/user-guides/alerting.md:40]:
```suggestion
The following example configuration creates an AlertmanagerConfig resource that sends notifications to a non-existent `wechat` receiver:
```

(removing the part of the sentence that suggests that something like this would be necessary to allow Alertmanager to start)

File: Documentation/user-guides/alerting.md
Comment (grdryn) [inline Documentation/user-guides/alerting.md:82]:
```suggestion
The `alertmanagerConfigSelector` field in the Alertmanager resource Spec needs to be specified so that the operator can select such AlertmanagerConfig resources. In the previous example, the label `alertmanagerConfig: example` is added, so the Alertmanager instance should be updated, adding the `alertmanagerConfigSelector`:
```

File: Documentation/user-guides/alerting.md
Comment (grdryn) [inline Documentation/user-guides/alerting.md:99]:
```suggestion
The following example configuration sends notifications to a non-existent `webhook`:
```

File: Documentation/user-guides/alerting.md
Comment (grdryn) [inline Documentation/user-guides/alerting.md:149]:
```suggestion
Once the operator merges the optional manually specified Secret with any selected `AlertmanagerConfig` resources, a new configuration Secret is created with the name `alertmanager-<Alertmanager name>-generated`, and is mounted into Alertmanager Pods created through the Alertmanager object.
```

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go
Comment (grdryn) [inline pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:172]: There are a couple of extra properties in the upstream type that are not reflected here:

* [HTTPConfig](https://github.com/prometheus/alertmanager/blob/master/config/notifiers.go#L397)
* [MessageType](https://github.com/prometheus/alertmanager/blob/master/config/notifiers.go#L407)

Do you plan to add those?

**Review by grdryn — COMMENTED** [review]
Thanks for updating the doc, it seems I completely missed that when adding the initial AlertmanagerConfig stuff! I've suggested a couple of changes to your proposed changes here.

File: Documentation/user-guides/alerting.md
Comment (zwpaper) [reply Documentation/user-guides/alerting.md:21]: how about changing this to

```
The AlertmanagerConfig resource is currently v1alpha1, testing and feedback are welcome.
```

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go
Comment (zwpaper) [reply pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:172]: my bad, I only checked the alertmanager doc website, not finding those two configs, added!

**Comment by zwpaper** [comment]
hi @grdryn , I have changed the doc and added the HTTPConfig and MessageType, thanks for the review.

after I click the re-request review from you, the reviewer `prometheus-operator-reviewer` is gone, is this as expected?

File: Documentation/user-guides/alerting.md
Comment (grdryn) [reply Documentation/user-guides/alerting.md]: Sure, that sounds better to me :+1: Will leave it up to the project maintainers to review and have the final say :slightly_smiling_face:

**Comment by grdryn** [comment]
@zwpaper thanks for addressing my comments, I don't have any more. Maybe one of the 

> after I click the re-request review from you, the reviewer prometheus-operator-reviewer is gone, is this as expected?

Hmm, I'm not sure, I don't know anything about that really. Maybe @simonpasquier or @lilic might know how to get that back if it's important? :slightly_smiling_face:

**Comment by zwpaper** [comment]
@grdryn It's not that important if the project maintainers can notice the PR😃

**Review by lilic — APPROVED** [review]
Very nice, thanks so much!!

lgtm

**Pull request merged**
